node/pkg/tss/util.go

```diff
@@ -0,0 +1,212 @@
+package tss
+
+import (
+	"fmt"
+
+	tsscommv1 "github.com/certusone/wormhole/node/pkg/proto/tsscomm/v1"
+	"github.com/yossigi/tss-lib/v2/ecdsa/signing"
+	"github.com/yossigi/tss-lib/v2/tss"
+	"go.uber.org/zap"
+)
+
+type logableError struct {
+	cause      error
+	trackingId []byte
+	round      signingRound // TODO: consider, it might require following the state of the fullParty per message.
+}
+
+func (l logableError) Error() string {
+	if l.cause == nil {
+		return ""
+	}
+	return l.cause.Error()
+}
+
+// Unwrap ensures logableError supports errors.Is and errors.As methods.
+func (l logableError) Unwrap() error {
+	return l.cause
+}
+
+func logErr(l *zap.Logger, err error) {
+	if l == nil {
+		return
+	}
+
+	if err == nil {
+		return
+	}
+
+	informativeErr, ok := err.(logableError)
+	if !ok {
+		l.Error(err.Error())
+		return
+	}
+
+	switch {
+	case informativeErr.round != "", informativeErr.trackingId != nil:
```

Comment:
This switch means informativeErr.round != "" or informativeErr.trackingId != nil, then
informativeErr.trackingId != nil is also covered in the next case

Reply:
my bad, I'll fix it.